dockstore-webservice/src/main/java/io/dockstore/webservice/resources/EventResource.java

```diff
@@ -99,24 +101,52 @@ public List<Event> getEvents(@Parameter(hidden = true) @ApiParam(hidden = true)
         @Min(1) @Max(MAX_LIMIT) @DefaultValue(PAGINATION_DEFAULT_STRING) @ApiParam(defaultValue = PAGINATION_DEFAULT_STRING, allowableValues = PAGINATION_RANGE) @Parameter(schema = @Schema(maximum = "100", minimum = "1")) @QueryParam("limit") Integer limit,
         @QueryParam("offset") @DefaultValue("0") Integer offset) {
         User userWithSession = this.userDAO.findById(user.getId());
-        return getEventsForUser(userWithSession, eventSearchType, limit, offset);
+        if (user.getIsAdmin() || user.isCurator()) {
+            return getEventsForUser(userWithSession, eventSearchType, limit, offset);
+        } else {
+            return filterCategoryEvents(getEventsForUser(userWithSession, eventSearchType, limit, offset));
```

Comment:
The problem with filtering like this is that you may end up with fewer events than `limit`. For example,

1. Limit is 100
2. `getEventsForUser` is invoked and returns 100 objects.
3. You then run `filterCategoryEvents` on the list from step 2, and, for example, 95 objects come back, even though there may be more than 100 for the user. Or hypothetically (I don't know the actual data very well), you could even have 0 objects come back from the filter, even though there are some events for the user, they're just not in the first batch.

I think you may need to tweak or add JPA queries.